lib/msal-browser/src/interaction_client/PopupClient.ts

```diff
@@ -546,58 +548,71 @@ export class PopupClient extends StandardInteractionClient {
         popupWindowParent: Window
     ): Promise<string> {
         return new Promise<string>((resolve, reject) => {
-            this.logger.verbose(
-                "PopupHandler.monitorPopupForHash - polling started"
-            );
-
-            const intervalId = setInterval(() => {
-                // Window is closed
-                if (popupWindow.closed) {
-                    this.logger.error(
-                        "PopupHandler.monitorPopupForHash - window closed"
-                    );
-                    clearInterval(intervalId);
+            setTimeout(() => {
```

Comment:
Do you need the setTimeout anymore? That was previously for polling changes to the popup's URI, but now you're getting an event from BoradcastChannel and IIRC there's no race - you should be able to hook up your message event handler immediately?